src/App.tsx

```diff
@@ -1,14 +1,84 @@
-/* eslint-disable max-len */
-import React from 'react';
+import {
+  FC, useCallback, useEffect, useMemo, useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+import { FilterBy } from './types/FilterBy';
+
+const findQueryInTheString = (query: string, title: string) => {
+  const queryToLower = query.toLowerCase().trim();
+  const titleToLower = title.toLowerCase();
+
+  return titleToLower.includes(queryToLower);
+};
+
+const debounce = (f: (string: string) => void, delay: number) => {
+  let timerId: any;
+
+  return (...args: any) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, ...args);
+  };
+};
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoadingTodos, setIsLoadingTodos] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [option, setOption] = useState<FilterBy>(FilterBy.all);
+
+  // eslint-disable-next-line no-console
+  console.log(option, appliedQuery);
```

Comment:
Remove the console.log statement, as it's not good practice to leave it in production code.